src/Framework/AssertionWarning.php

```diff
@@ -0,0 +1,27 @@
+<?php
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+/**
+ * Thrown when an assertion warns.
+ *
+ * @since Class available since Release 5.0.0
+ */
+class PHPUnit_Framework_AssertionWarning extends PHPUnit_Framework_Exception implements PHPUnit_Framework_SelfDescribing
```

Comment:
The point is/was to get rid of the existing `PHPUnit_Framework_Warning extends PHPUnit_Framework_TestCase` ugliness :-)